File: modules/daily-task-notifications/src/DailyTaskNotificationsModule.ts
import { NativeModule, requireNativeModule } from 'expo';
import type {
  DailyTaskNotificationsModuleEvents,
} from './DailyTaskNotifications.types';

declare class DailyTaskNotificationsModule extends NativeModule<DailyTaskNotificationsModuleEvents> {
  showDailyTaskNotification(notificationJson: string): Promise<void>;
  updateDailyTaskNotification(notificationJson: string): Promise<void>;
  cancelDailyTaskNotification(notificationKey: string): Promise<void>;
  getPendingActionsAsync(): Promise<string>;
  removePendingActionsAsync(actionIds: string[]): Promise<void>;
}

export default requireNativeModule<DailyTaskNotificationsModule>('DailyTaskNotifications');

File: modules/daily-task-notifications/ios/DailyTaskNotificationsModule.swift
import ExpoModulesCore

public class DailyTaskNotificationsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("DailyTaskNotifications")
    Events("onActionPressed")

    AsyncFunction("showDailyTaskNotification") { (_ notificationJson: String) in
      return
    }

    AsyncFunction("updateDailyTaskNotification") { (_ notificationJson: String) in
      return
    }

    AsyncFunction("cancelDailyTaskNotification") { (_ notificationKey: String) in
      return
    }

    AsyncFunction("getPendingActionsAsync") {
      return "[]"
    }

    AsyncFunction("removePendingActionsAsync") { (_ actionIds: [String]) in
      return
    }
  }
}
